gillespy2/solvers/cpp/c_base/tau_hybrid_cpp_solver/TauHybridSolver.cpp

```diff
@@ -0,0 +1,216 @@
+					IntegrationResults result = sol.integrate(&next_time);
+
+					// 0-initialize our population_changes array.
+					for (int p_i = 0; p_i < num_species; ++p_i) {
+						population_changes[p_i] = 0;
+					}
+
+					// Start with the species concentration as a baseline value.
+					// Stochastic reactions will update populations relative to their concentrations.
+					for (int spec_i = 0; spec_i < num_species; ++spec_i) {
+						current_state[spec_i] = result.concentrations[spec_i];
+					}
+
+					// The newly-updated reaction_states vector may need to be reconciled now.
+					// A positive reaction_state means reactions have potentially fired.
+					// NOTE: it is possible for a population to swing negative, where a smaller Tau is needed.
+					for (int rxn_i = 0; rxn_i < num_reactions; ++rxn_i) {
+						// Temporary variable for the reaction's state.
+						// Does not get updated unless the changes are deemed valid.
+						double rxn_state = result.reactions[rxn_i];
+
+						switch (simulation->reaction_state[rxn_i].mode) {
+						case SimulationState::DISCRETE:
+							while (rxn_state >= 0) {
+								// "Fire" a reaction by recording changes in dependent species.
+								// If a negative value is detected, break without saving changes.
+								for (int spec_i = 0; spec_i < num_species; ++spec_i) {
+									population_changes[spec_i] +=
+										model.reactions[rxn_i].species_change[spec_i];
+									if (current_state[spec_i] + population_changes[spec_i] < 0) {
+										invalid_state = true;
+									}
+								}
+
+								rxn_state += log(urn.next());
+							}
+							result.reactions[rxn_i] = rxn_state;
+							break;
+
+						case SimulationState::CONTINUOUS:
+						default:
+							break;
+						}
+					}
+
+					// Positive reaction state means a negative population was detected.
+					// Only update state with the given population changes if valid.
+					if (invalid_state) {
+						sol.restore_state();
+						tau_step /= 2;
+						next_time = simulation->current_time + tau_step;
+					}
+					else {
+						// "Permanently" update the rxn_state and populations.
+						for (int p_i = 0; p_i < num_species; ++p_i) {
+							current_state[p_i] += population_changes[p_i];
+							current_populations[p_i] = current_state[p_i];
+							result.concentrations[p_i] = current_state[p_i];
+						}
+					}
+				} while (invalid_state);
+
+				// Output the results for this time step.
+				sol.refresh_state();
+				simulation->current_time = next_time;
+
+				// Seek forward, writing out any values on the timeline which are on current timestep range.
+				// 
+				while (save_idx < simulation->number_timesteps && save_time <= next_time) {
+					for (int spec_i = 0; spec_i < num_species; ++spec_i) {
+						simulation->trajectories[traj][save_idx][spec_i] = current_state[spec_i];
+					}
+					save_time = simulation->timeline[++save_idx];
+				}
+			}
+
+			// End of trajectory
+			delete[] population_changes;
+		}
+	}
+}
```

Comment:
Same thing here -- file needs a formatting pass to ensure it follows codebase style.